sdk/python/feast/infra/gcp.py

```diff
@@ -104,6 +105,24 @@ def teardown_infra(
             key = client.key("Project", project, "Table", table.name)
             client.delete(key)
 
+
+class OnlineStoreDatastore(OnlineStore):
+    _gcp_project_id: Optional[str]
+
+    def __init__(self, config: Optional[DatastoreOnlineStoreConfig]):
+        if config:
+            self._gcp_project_id = config.project_id
+        else:
+            self._gcp_project_id = None
+
+    def _initialize_client(self):
```

Comment:
It's not really clear to me what purpose the `initialize_client` serves? Could this logic be moved to the constructor? Also, is there a reason the `from google.cloud import datastore` import is called twice? Here, and also at the call site/

Reply:
It is juts an utility function, I generally try not to have constructor to do any complex initialization that can fail. Not sure how to avoid the import? Both sites need this module but I don't want to do a top level import to avoid hard dependency.